nasa/harmony-gdal-adapter
# HGA migrate staticmethods out of HarmonyAdapter

## Description

This PR is the continuing adventure of cleaning up HGA. It does 3 main things (roughly split by pre-squash-merge commits):

1) A few more linting issues (mainly whitespace related).
2) Removes the `HarmonyAdapter.cmd2` and `HarmonyAdapter.cmd3` methods. `HarmonyAdapter.cmd2` has been removed in favour of Python-native functions from `os` and `shutil` to copy, move, delete files/directories. `HarmonyAdapter.cmd3` was used in one place and has been removed in favour of an alternative way to retrieve `gdalinfo` output.
3) Migrating staticmethods from `HarmonyAdapter` into utility Python modules. I've made a couple of minor tweaks as I go (some function names are slightly different, and I've added some typing), but generally, I've aimed to primarily move the methods with a view to updating them (and adding unit tests) in future PRs.

As with the previous clean-up PR - there's definitely still a lot of clean-up to do. I'm sure I've missed a few things. The idea is to keep at this in an iterative fashion. If I've missed something super egregious, though, let me know.

## Jira Issue ID

N/A

## Local Test Steps

* Pull this branch of HGA locally.
* Build the HGA service Docker image from this branch (`bin/build-image`).
* Also build and run the test Docker image (`bin/build-test` and `bin/run-test`). The tests should all pass.
* Restart the services in your local Harmony instance, to start using the updated HGA image from this branch.
* Run the HGA regression suite, updating the first cell to: `harmony_host_url = 'http://localhost:3000'`. All the tests should pass.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~Documentation updated (if needed).~~
